sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -470,7 +497,20 @@ def test_error_on_missing_requests_for_sub_estimator(metaestimator):
             instance = cls(**kwargs)
             with pytest.raises(UnsetMetadataPassedError, match=re.escape(msg)):
                 method = getattr(instance, method_name)
-                method(X, y, **method_kwargs)
+                if method_name in ["predict", "score"]:
+                    # set request on fit and on the method not tested here
+                    set_request(estimator, "fit")
+                    if method_name == "predict":
+                        set_request(estimator, "score")
+                    if method_name == "score":
+                        set_request(estimator, "predict")
+                    # fit before calling method
+                    fit_method = getattr(instance, "fit")
+                    fit_method(X, y, **method_kwargs)
+                if method_name == "predict":
+                    method(X, **method_kwargs)
+                else:
+                    method(X, y, **method_kwargs)
```

Comment:
this can be simplified, a single call to `set_request`, and only a check if `fit` is in `method_name` or not, and call fit before the method.